mono/metadata/object-internals.h

```diff
@@ -168,7 +168,7 @@ struct _MonoString {
 
 // Equivalent to mono_array_addr_with_size, except:
 // 1. A macro instead of a function.
-// 2. No GC enter/exit unsafe transition.
+// 2. mono_array_addr_with_size could, but does not, do GC mode transitions.
```

Comment:
The macro vs. function means the types  of the parameters are "generic".
Or we could cast them. Or review the few uses. Or change it to an inline function.
Or make an _internal form that is inline function, for extreme parity.

Reply:
It looks like the only? two uses had type parity already.

Reply:
For size to be a signed int is kinda bogus, should be unsigned size_t.